packages/trader/src/App/Components/Form/RangeSlider/range-slider.jsx

```diff
@@ -6,82 +6,128 @@ import { localize } from 'App/i18n';
+    }
+
+    handleChange = (e) => {
+        if (e.target.value !== this.props.value) {
+            this.resetOnHover();
+            this.props.onChange({ target: { name: this.props.name, value: e.target.value } });
         }
     };
 
-    const handleClick = (e, index) => {
-        if (index !== value) {
-            onChange({ target: { name, value: index } });
+    handleClick = (e, index) => {
+        if (index !== this.props.value) {
+            this.resetOnHover();
+            this.props.onChange({ target: { name: this.props.name, value: index } });
         }
     };
 
-    if (+value < min_value || +value > max_value) {
-        onChange({
-            target: {
-                name,
-                value: min_value,
-            },
-        });
-    }
+    onMouseEnter = (e, index) => {
+        if (index) {
+            this.setState({ hover_value: index });
+            this.rangeSliderTrack.style.width = this.getRangeSliderTrackWidth(index, true);
+        }
+    };
+
+    onMouseLeave = (e) => {
+        const { offsetX, offsetY } = e.nativeEvent;
+        if (offsetY <= -3 || offsetY >= 3 || offsetX < -3 || offsetX > 3) {
+            this.resetOnHover();
+        }
+    };
+
+    getRangeSliderTrackWidth = (value, is_hover) => {
+        const width = (value - this.props.min_value) * (10 / (this.props.max_value - this.props.min_value));
+        return `${(width * 2) + (is_hover ? 0.8 : 1.4)}em`;
+    };
 
-    return (
-        <div className={classNames('range-slider', className, { 'range-slider__error': ((value < +min_value) || (value > +max_value)) })}>
-            <label className='range-slider__label' htmlFor='range'>
-                <input
-                    id='range'
-                    className='input trade-container__input range-slider__track'
-                    type='range'
-                    min={min_value}
-                    max={max_value}
-                    min_value={min_value}
-                    max_value={max_value}
-                    name={name}
-                    steps={max_value - min_value}
-                    onChange={handleChange}
-                    tabIndex='0'
-                    value={value}
-                />
-                <div className='range-slider__ticks'>
-                    <TickSteps
+    resetOnHover = () => {
+        if (this.state.hover_value) {
+            this.setState({ hover_value: 0 });
+            this.rangeSliderTrack.style.width = 0;
+        }
+    };
+
+    render() {
+        const {
+            className,
+            max_value,
+            min_value,
+            name,
+            value,
+            onChange,
+        } = this.props;
+
+        if (+value < min_value || +value > max_value) {
+            onChange({
+                target: {
+                    name,
+                    value: min_value,
+                },
+            });
+        }
+
+        const display_value = this.state.hover_value || value;
```

Comment:
Maybe u can add `+` around `display_value` here if you want it to always return a number? 
Instead of adding it on L119 and L120.

Reply:
good point 👍